### PR TITLE
Add idiom on resource-manager api

### PR DESCRIPTION
**Pull request by jcrist**
Adds an idiom on using the `resource_manager` api, with an example using a temporary dask cluster.

### CODE REVIEW

File: docs/core/idioms/resource-manager.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+# Managing temporary resources
+
+Sometimes you have workloads that depend on temporary that need to be setup,
```
Comment (cicdw) [inline docs/core/idioms/resource-manager.md:3]: temporary -- ?

File: docs/core/idioms/resource-manager.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    def cleanup(self, resource):
+        """Cleanup the resource.
+
+        This receives the result of `setup`, and is always called if `setup`
+        succeeds, even if other upstream tasks failed.
+        """
+        ...
+```
+
+The resulting `ResourceManager` can then be used in a `Flow` as a
```
Comment (cicdw) [inline docs/core/idioms/resource-manager.md:50]:
```suggestion
The resulting `ResourceManager` can then be used when building a `Flow` as a
```

File: docs/core/idioms/resource-manager.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+flow = Flow("dask-example")
+n_workers = Parameter("n_workers", default=None)
+out_path = Parameter("out_path", default="summary.csv")
+
+with DaskCluster(n_workers=n_workers, flow=flow) as client:
+    # These tasks rely on a dask cluster to run, so we create them inside
+    # the `DaskCluster` resource manager
+    df = LoadData()
+    summary = Summarize().bind(df, client, flow=flow)
```
Comment (cicdw) [inline docs/core/idioms/resource-manager.md:178]: Interesting, for some reason I've never considered `bind` a part of the imperative API but it makes sense!

**Comment by jcrist** [comment]
I believe all issues have been addressed.

**Pull request merged**